Could not build SFM

As per the title,

the build process stops with the following error:

```
[ 91%] Building CXX object modules/SFM/CMakeFiles/SFM.dir/SFM.cpp.o
In file included from /home/xenvre/stereo-vision/modules/SFM/SFM.h:250,
                 from /home/xenvre/stereo-vision/modules/SFM/SFM.cpp:22:
/home/xenvre/stereo-vision/modules/SFM/fastBilateral.hpp: In function ‘void cv_extend::bilateralFilterImpl(cv::Mat1d, cv::Mat1d, double, double)’:
/home/xenvre/stereo-vision/modules/SFM/fastBilateral.hpp:126:24: warning: narrowing conversion of ‘(std::size_t)small_height’ from ‘std::size_t’ {aka ‘long unsigned int’} to ‘int’ [-Wnarrowing]
  126 |     int data_size[] = {small_height, small_width, small_depth};
      |                        ^~~~~~~~~~~~
/home/xenvre/stereo-vision/modules/SFM/fastBilateral.hpp:126:38: warning: narrowing conversion of ‘(std::size_t)small_width’ from ‘std::size_t’ {aka ‘long unsigned int’} to ‘int’ [-Wnarrowin ]
  126 |     int data_size[] = {small_height, small_width, small_depth};
      |                                      ^~~~~~~~~~~
/home/xenvre/stereo-vision/modules/SFM/fastBilateral.hpp:126:51: warning: narrowing conversion of ‘(std::size_t)small_depth’ from ‘std::size_t’ {aka ‘long unsigned int’} to ‘int’ [-Wnarrowin ]
  126 |     int data_size[] = {small_height, small_width, small_depth};
      |                                                   ^~~~~~~~~~~
/home/xenvre/stereo-vision/modules/SFM/SFM.cpp: In member function ‘void SFM::fillWorld3D(yarp::sig::ImageOf<yarp::sig::PixelRgbFloat>&, yarp::sig::ImageOf<yarp::sig::PixelRgbFloat>&)’:
/home/xenvre/stereo-vision/modules/SFM/SFM.cpp:1324:45: error: conversion from ‘const long int’ to non-scalar type ‘CvScalar’ requested
 1324 |             CvScalar scal = disp16m.at<long>(v_, u_);
      |                             ~~~~~~~~~~~~~~~~^~~~~~~~
make[2]: *** [modules/SFM/CMakeFiles/SFM.dir/build.make:76: modules/SFM/CMakeFiles/SFM.dir/SFM.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:202: modules/SFM/CMakeFiles/SFM.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

The error seems to be due to the following conversion:

https://github.com/robotology/stereo-vision/blob/8b61da4ce6080e792f17c12c91fc23fda903fdae/modules/SFM/SFM.cpp#L1324

That line of code was changed in a recent PR (see https://github.com/robotology/stereo-vision/pull/30/files).

I am running `Ubuntu 20.04.2` with `OpenCV 4.2.0` compiled from source in order to enable non-free SIFT features.

--- a/modules/SFM/SFM.cpp
+++ b/modules/SFM/SFM.cpp
@@ -19,6 +19,7 @@
 #include <cmath>
 #include <algorithm>
 #include <yarp/cv/Cv.h>
+#include <opencv2/core/types.hpp>
 #include "SFM.h"
 
 using namespace yarp::cv;
@@ -1321,7 +1322,7 @@ void SFM::fillWorld3D(ImageOf<PixelRgbFloat> &worldCartImg,
             y=(vsign+1)*Q.at<double>(1,1)+Q.at<double>(1,3);
             z=Q.at<double>(2,3);
 
-            CvScalar scal = disp16m.at<long>(v_, u_);
+            cv::Scalar scal(disp16m.at<long>(v_, u_));
 
             double disparity=scal.val[0]/16.0;
             double w=disparity*Q.at<double>(3,2)+Q.at<double>(3,3);

